docs/5.0/kubernetes-access.md

```diff
@@ -5,61 +5,244 @@ description: How to set up and configure Teleport for Kubernetes access with SSO
+    Forward ports to provide UI and CLI access.
+    ```
+    # Change teleport-699d68645c-wpqm7 to the name of the Pod
+    kubectl port-forward teleport-699d68645c-wpqm7 6379:32334
+    ```
+
+    Next Step: Create a Teleport User & Setup 2FA
+    ```
+    kubectl exec deployments/teleport -- tctl users add testuser root,ubuntu,k8s
+    ```
+
+
+=== "Enterprise"
+
+    Prepare a Teleport Enterprise license file<br>
+    If you are deploying the Enterprise version you will require the license file as a secret available to Teleport.
+    Download the `license.pem` from the [Teleport dashboard,](https://dashboard.gravitational.com/web/login) and rename
+    it to the filename that this chart expects:
+
+    ```bash
+    cp ~/Downloads/license.pem license-enterprise.pem
+    ```
+
+    Store it as a Kubernetes secret:
+    ```
+    kubectl create secret generic license --from-file=license-enterprise.pem
+    ```
+
+    Setup and Install Teleport Helm chart repository.
+    ```bash
+    helm repo add teleport https://charts.releases.teleport.dev
+    helm install teleport teleport/teleport
+    ```
+
+    Download Demo Helm Value and Update Teleport Pod.
+    ```bash
+    curl https://github.com/gravitational/teleport/blob/master/examples/chart/teleport/values.yaml  --output values.yaml
+    helm upgrade -f values.yaml teleport teleport/teleport
+    ```
+
+    Check that the Teleport Service is running.
+    ```bash
+    kubectl get service
+    # NAME       TYPE       CLUSTER-IP    EXTERNAL-IP   PORT(S)                                                                      AGE
+    # teleport   NodePort   10.20.28.76   <none>        3025:32676/TCP,3026:30207/TCP,3023:32763/TCP,3024:31710/TCP,3080:32333/TCP   89m
+    ```
+
+    !!! Tip
+
+        If Teleport isn't starting you can tail Kubernetes logs for more info `kubectl logs --follow deployments/teleport`
+        or `kubectl get pods`<br>_Example: `kubectl describe pod teleport-5f5f989b96-9khzq`_
+
+    Forward ports to provide UI and CLI access.
+    ```
+    # Change teleport-699d68645c-wpqm7 to the name of the Pod
+    kubectl port-forward teleport-699d68645c-wpqm7 6379:32334
+    ```
+
+    Create a Teleport User & Setup 2FA
+    ```
+    # tctl is a tool to administer Teleport
+    kubectl exec deployments/teleport -- tctl users add testuser root,ubuntu,k8s
+    ```
+
+Teleport is now setup within Kubernetes. This example provides a simple POC for
+deploying Teleport. For Production deploys you should [Add TLS Certificates](https://github.com/gravitational/teleport/tree/master/examples/chart/teleport#adding-tls-certificates), consider setting up for [High Availability](https://github.com/gravitational/teleport/blob/master/examples/chart/teleport/HIGHAVAILABILITY.md) and changing Teleports service type to an option such
+as [LoadBalancer](https://github.com/gravitational/teleport/blob/master/examples/chart/teleport/values.yaml#L197)
+
+
+## Deploy Teleport outside Kubernetes
```

Comment:
We are mixing two use cases here in one. Let's split into two chapters:

1. Connect Kuberentes cluster to externa Teleport Access Plane

Explain in one sentence why (Users can connect multiple Kubernetes clusters to the same access plane.)
Show the diagram that connects multiple K8s clusters to the access plane.

2. Host Teleport Cluster in front of Kubernetes

This is a niche use case for existing clusters of for some hybrid scenarios. Explain the need for it an have a separate flow.

Reply:
I'm a little unclear on the 2nd point. `Host Teleport Cluster in front of Kubernetes`: Is this referring to running the root completely outside of Kubernetes e.g. Teleport Cloud -> Multiple Clusters?

Reply:
`Deploy Teleport outside Kubernetes` means that teleport is running in front of one or many K8s clusters required for some hybrid scenarios. This is different from Kubernetes agent dialing back the cluster scenario 1.